src/test/java/com/marklogic/kafka/connect/source/AbstractIntegrationSourceTest.java

```diff
@@ -117,4 +119,24 @@ private void assertTopicAndSingleValue(List<SourceRecord> newSourceRecords, Stri
     protected Stream<ObjectNode> recordsToJsonObjects(List<SourceRecord> records) {
         return records.stream().map(record -> readJsonObject((String) record.value()));
     }
+
+    protected String injectValueIntoQuery(String userDsl, Map<String, Object> parsedConfig, String CONSTRAINT_VALUE) {
+        parsedConfig.put(MarkLogicSourceConfig.DSL_QUERY, userDsl);
+        return new DslQueryHandler(null, parsedConfig).injectConstraintIntoQuery(CONSTRAINT_VALUE);
+    }
+
+    protected void verifyQueryReturnsExpectedRows(String constraintValue, int expectedRowCount, String stringInFirstRecord, Map<String, Object> parsedConfig) {
+        QueryHandler queryHandler = QueryHandler.newQueryHandler(getDatabaseClient(), parsedConfig);
+        PlanBuilder.Plan plan = queryHandler.newPlan(constraintValue);
+        PlanInvoker.Results results = PlanInvoker.newPlanInvoker(getDatabaseClient(), parsedConfig).invokePlan(plan, "Authors");
+        List<SourceRecord> newRecords = results.getSourceRecords();
+        assertEquals(expectedRowCount, newRecords.size());
+        assertTrue(((String) newRecords.get(0).value()).contains(stringInFirstRecord));
```

Comment:
Whenever assertTrue or assertFalse is used, it's good to include a message so that when the test fails, we know what the actual value is (as JUnit won't tell us anything other than "assertion failed"). And in a helper method like this that's used across tests, it's helpful to include the value that was passed in. So do:

    String firstRecordValue = (String)newRecords.get(0).value();
    assertTrue(firstRecordValue.contains(stringInFirstRecord), 
    "Did not find " + stringInFirstRecord " + " in " + firstRecordValue);